pkg/infra/usagestats/statscollector/prometheus_flavor.go

```diff
@@ -44,35 +47,59 @@ func (s *Service) detectPrometheusVariants(ctx context.Context) (map[string]int6
 		return nil, err
 	}
 
-	variants := map[string]int64{}
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
Nit. You might want to set a concurrency limit here in case there are A LOT OF prometheus datasources, e.g. `g.SetLimit(<limit>)`